File: 백준/Silver/1629. 곱셈/곱셈.java
import java.io.*;
import java.util.*;

public class Main {
    static long A, B, C;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        A = Long.parseLong(st.nextToken());
        B = Long.parseLong(st.nextToken());
        C = Long.parseLong(st.nextToken());

        System.out.print(dnc(B));
    }
    static long dnc(long idx) {
        if (idx == 1) return A%C; // 재귀 탈출 조건
        long tmp = dnc(idx/2);
        return ((idx%2==0? 1:A%C) * (tmp * tmp % C))%C;
    }
}

File: 백준/Silver/1629. 곱셈/곱셈.py
import sys
input = sys.stdin.readline

A, B, C = map(int, input().split())

def dnc(idx):
    if idx == 1 : return A % C
    return ((A if idx%2 else 1) * (dnc(idx//2)**2))% C

print(dnc(B))